src/stor/Context.tsx

```diff
@@ -0,0 +1,262 @@
+import React, {
+  createRef,
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Todo } from '../types/Todo';
+import * as postService from '../api/todos';
+
+enum FilerType {
+  FILTER_TODO_ALL = 'all',
+  FILTER_TODO_ACTIVE = 'active',
+  FILTER_TODO_COMPLETED = 'completed',
+}
+
+type TodosContextType = {
+  todos: Todo[];
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  toggleAll: () => void;
+  addTodo: (newTodo: Todo) => Promise<void>;
+  isAllTodoCompleted: boolean;
+  updateTodo: (newTodo: Todo) => Promise<void>;
+  deleteTodo: (deletedTodo: Todo) => void;
+  clearCompleted: () => void;
+  filterField: FilerType;
+  setFilterField: React.Dispatch<React.SetStateAction<FilerType>>;
+  visibleTodos: Todo[];
+  todoRef: React.RefObject<HTMLInputElement>;
+  errorMessage: string;
+  setErrorMessage: React.Dispatch<React.SetStateAction<string>>;
+  stateClearBtn: boolean;
+  setStateClearBtn: React.Dispatch<React.SetStateAction<boolean>>;
+  isSubmitting: boolean;
+  setIsSubmitting: React.Dispatch<React.SetStateAction<boolean>>;
+  tempTodo: Todo | null;
+  loadingIds: number[];
+};
+
+export const TodosContext = React.createContext<TodosContextType>({
+  todos: [],
+  setTodos: () => {},
+  toggleAll: () => {},
+  addTodo: async () => {},
+  isAllTodoCompleted: false,
+  updateTodo: async () => {},
+  deleteTodo: () => {},
+  clearCompleted: () => {},
+  filterField: FilerType.FILTER_TODO_ALL,
+  setFilterField: () => {},
+  visibleTodos: [],
+  todoRef: createRef(),
+  errorMessage: '',
+  setErrorMessage: () => {},
+  stateClearBtn: false,
+  setStateClearBtn: () => {},
+  isSubmitting: false,
+  setIsSubmitting: () => {},
+  tempTodo: null,
+  loadingIds: [],
+});
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterField, setFilterField] = useState(FilerType.FILTER_TODO_ALL);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [stateClearBtn, setStateClearBtn] = useState(true);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+
+  const todoRef = useRef<HTMLInputElement>(null);
+
+  const addTodo = useCallback(({ title, completed, userId }: Todo) => {
+    setErrorMessage('');
+
+    if (title.trim() !== '') {
+      setTempTodo({ id: 0, title, completed, userId });
+    }
+
+    return postService
+      .createTodos({ title, completed, userId })
+      .then(newTodo => {
+        setTodos(curentTodos => [...curentTodos, newTodo]);
+      })
+      .catch(error => {
+        setErrorMessage('Unable to add a todo');
+        setTimeout(() => setErrorMessage(''), 3000);
+        throw error;
```

Comment:
do not throw an error here - you handled it already, so no need to throw it again. It will lead to a page crash (because you don't try to catch the error thrown here somewhere else)